src/libraries/System.Net.Security/tests/FunctionalTests/SslAuthenticationOptionsTest.cs

```diff
@@ -74,9 +76,9 @@ public async Task ClientOptions_ServerOptions_NotMutatedDuringAuthentication()
                     };
 
                     // Authenticate
-                    Task clientTask = client.AuthenticateAsClientAsync(clientOptions, default);
-                    Task serverTask = server.AuthenticateAsServerAsync(serverOptions, default);
-                    await new[] { clientTask, serverTask }.WhenAllOrAnyFailed();
+                    Task clientTask = client.AuthenticateAsClientAsync(clientOptions, async);
```

Comment:
I wish it is somehow more visible this calls wrapper since there is no AuthenticateAsClientAsync() overload taking Options and bool. It is not obvious from first look what Sio going on IMHO.

Reply:
I have a two options for you:
- rename the extension to `AuthenticateAsClientAsyncEx` or `AuthenticateAsClientAsyncTest`
- don't call it as an extension but rather explicitly: `SslStreamExtensions.AuthenticateAsClientAsync(client, clientOptions, async)`

If you think it's worth pursuing, choose 😄

Reply:
I would probably prefer AuthenticateAsClientAsyncEx() (or something like that) but I think it is ok to keep it as is unless somebody else asks for it.

Reply:
I just changed the names and order of args (`client.AuthenticateAsClientAsync(TestAuthenticateAsync, clientOptions)`). Hopefully, it's more obvious this is intended for tests only.
I'm following the same pattern I'm using in the sync `HttpClient` api so in case somebody expresses the same concern as you, I'll go back and address it here as well.